src/audio_call.c

```diff
@@ -389,74 +421,67 @@ void callback_recv_invite(Tox *m, uint32_t friend_number)
 
     for (uint8_t i = 0; i < MAX_WINDOWS_NUM; ++i) {
         if (windows[i] != NULL && windows[i]->onInvite != NULL && windows[i]->num == friend_number) {
-            windows[i]->onInvite(windows[i], CallControl.av, friend_number, CallControl.call_state);
+            windows[i]->onInvite(windows[i], CallControl.av, friend_number, CallControl.calls[friend_number].state);
```

Comment:
Given that `CallControl.calls[friend_number]` is invariant, you could hoist it out of the loop here and in all other similar loops below.